server/interpreter.js

```diff
@@ -2274,48 +2242,48 @@ Interpreter.prototype['stepAssignmentExpression'] = function() {
     this.pushNode_(node['right']);
     return;
   }
+  var leftSide = state.leftValue_;
   var rightSide = state.value;
-  var value;
-  if (node['operator'] === '=') {
-    value = rightSide;
-  } else {
-    var rightValue = rightSide;
-    var leftNumber = state.leftValue_ - 0;
-    var rightNumber = rightValue - 0;
-    if (node['operator'] === '+=') {
-      var left, right;
-      if (typeof state.leftValue_ === 'string' ||
-          typeof rightValue === 'string') {
-        left = state.leftValue_ + '';
-        right = rightValue + '';
-      } else {
-        left = leftNumber;
-        right = rightNumber;
-      }
-      value = left + right;
-    } else if (node['operator'] === '-=') {
-      value = leftNumber - rightNumber;
-    } else if (node['operator'] === '*=') {
-      value = leftNumber * rightNumber;
-    } else if (node['operator'] === '/=') {
-      value = leftNumber / rightNumber;
-    } else if (node['operator'] === '%=') {
-      value = leftNumber % rightNumber;
-    } else if (node['operator'] === '<<=') {
-      value = leftNumber << rightNumber;
-    } else if (node['operator'] === '>>=') {
-      value = leftNumber >> rightNumber;
-    } else if (node['operator'] === '>>>=') {
-      value = leftNumber >>> rightNumber;
-    } else if (node['operator'] === '&=') {
-      value = leftNumber & rightNumber;
-    } else if (node['operator'] === '^=') {
-      value = leftNumber ^ rightNumber;
-    } else if (node['operator'] === '|=') {
-      value = leftNumber | rightNumber;
-    } else {
+  var value = leftSide;
+  switch (node['operator']) {
+    case '=':
+      value = rightSide;
+      break;
+    case '+=':
+      value += rightSide;
+      break;
+    case '-=':
```

Comment:
In this case I would be totally O.K. with ignoring the style guide and formatting these as:

```JS
  switch (node['operator']) {
    case '=':   value =   rightSide; break;
    case '+=':  value +=  rightSide; break;
```

I actually think it might actually be preferable because it would make it really obvious if there were any missing break statements, mismatched operators, variable misspellings, etc.